### PR TITLE
taskresource/cgroup: remove task dependency

### PR DESCRIPTION
**Pull request by sharanyad**
### Summary
Minor refactor - Remove task dependency for resource creation.

### Implementation details
Changing method `Create(task *api.Task) error`  to `Create() error`. `api.Task` would contain list of resources in the future. Hence, removing dependency of task in resources package. 
Passing all the fields needed for resource creation to `CgroupResource` object. 

### Testing
- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: yes

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/taskresource/cgroup/cgroup.go [hunk 1]
```diff
@@ -54,11 +56,13 @@ type CgroupResource struct {
 }
 
 // NewCgroupResource is used to return an object that implements the Resource interface
-func NewCgroupResource(control cgroup.Control, cgroupRoot string, cgroupMountPath string) *CgroupResource {
+func NewCgroupResource(taskArn string, control cgroup.Control, cgroupRoot string, cgroupMountPath string, resourceSpec specs.LinuxResources) *CgroupResource {
```
Comment (aaithal) [inline agent/taskresource/cgroup/cgroup.go]: minor: can you split this into multiple lines?

File: agent/taskresource/cgroup/cgroup.go [hunk 2]
```diff
@@ -42,9 +42,11 @@ var (
 
 // CgroupResource represents Cgroup resource
 type CgroupResource struct {
+	taskArn             string
```
Comment (aaithal) [inline agent/taskresource/cgroup/cgroup.go]: lint: taskARN

File: agent/taskresource/cgroup/cgroup.go [hunk 3]
```diff
@@ -22,11 +22,11 @@ import (
 	"sync"
 	"time"
 
-	"github.com/aws/amazon-ecs-agent/agent/api"
 	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
 	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
 	"github.com/cihub/seelog"
 	"github.com/containerd/cgroups"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
```
Comment (aaithal) [inline agent/taskresource/cgroup/cgroup.go]: Is this a new dependency? Does this require a dep update?

Comment (adnxn) [inline agent/taskresource/cgroup/cgroup.go]: looks like we track a [revision](https://github.com/aws/amazon-ecs-agent/blob/master/agent/Gopkg.toml#L62) for this dependency. since they have [versioned releases](https://github.com/opencontainers/runtime-spec/releases), we can track one of those. although this could be a separate PR against dev i guess? but ya, we should do this.

Comment (petderek) [reply agent/taskresource/cgroup/cgroup.go]: we were using this already in the cgroup codebase

Comment (sharanyad) [reply agent/taskresource/cgroup/cgroup.go]: This does need update. As @adnxn mentioned, we can make it track the latest release, which I shall do in another PR.
